Merge gh-pages-dev-2020.3 into gh-pages (#576)
tableau/connector-plugin-sdk: docs/auth-modes.md

Below is an example of how to pass DelegationUID in a JDBC plugin in Tableau Server. Please note that some JDBC drivers may have a different name for this property so take a look at driver documentation for the appropriate property name.
 
 ```javascript
    // Connection properties
    function isEmpty(str) {
        return (!str || 0 === str.length); 
    }

    var props = {};
    props["UID"] = attr[connectionHelper.attributeUsername];
    props["PWD"] = attr[connectionHelper.attributePassword];
   
    if (attr[connectionHelper.attributeTableauServerAuthMode] == connectionHelper.valueAuthModeDBImpersonate) {
        var str = attr[connectionHelper.attributeTableauServerUser];
        
        if (!isEmpty(str)){
            props["DelegationUID"] = str;
        }
    }
 ```
 

>  Also add CAP_AUTH_DB_IMPERSONATE to the list of capabilities in plugin manifest so that the option "Impersonate using embedded password" shows up in the publish dialog.

Reviewer: ```suggestion
To enable the "Impersonate using embedded password" option in the publish dialog ensure CAP_AUTH_DB_IMPERSONATE is enabled in the plugin manifest.xml.
```

 ```xml 
   // manifest.xml 
    <customizations>
      <customization name="CAP_AUTH_DB_IMPERSONATE" value="yes"/>
    </customizations>
 ```
 

> The connector plugin present on a Tableau Server instance MUST contain the above two code changes and the server admin should run "tsm pending-changes apply". After this, user should be able to use a vanilla plugin on a Tableau Desktop Client with the same class name and capable of doing basic username-password authentication. Thereafter, when the user tries to publish a workbook using Tableau Desktop to a Tableau Server instance, they should be able to publish it using the option "Impersonate using embedded password". When the workbook is opened in Tableau Server, the delegation should have taken place automatically if appropriate delegation permissions are configured for the authenticated_user and delegated_user(user logged into Tableau server) or else an error message of this sort is thrown from the database "User user1 is not authorized to delegate to user2". For more information on deploying connector plugins in Tableau Server, please refer this [document](https://tableau.github.io/connector-plugin-sdk/docs/share)

Reviewer: Is this saying we require two different versions of the plugin, one for server and one for desktop?  If not my preference would be they match.

I think we might be able to remove the part about how to install a plugin on server as well.

Maybe this is something as a follow-up item.

Author: I think it's one connector for both, at least we have one sample. @devanshsoni9  can you confirm?

> We are providing a [sample](https://github.com/tableau/connector-plugin-sdk/tree/master/samples/plugins/db_impersonation) JDBC plugin for the plugin developers using an Impala database as an example. For documentation on how to configure delegation in Impala, refer to database [documentation](https://impala.apache.org/docs/build/html/topics/impala_delegation.html).

Reviewer: ```suggestion
A [sample](https://github.com/tableau/connector-plugin-sdk/tree/master/samples/plugins/db_impersonation) JDBC plugin is provided using an Impala database as an example. For documentation on how to configure delegation in Impala, refer to database [documentation](https://impala.apache.org/docs/build/html/topics/impala_delegation.html).
```
